:thinking:  Python 2.7 failed because
```plain
Traceback (most recent call last):
  File "tests.py", line 42, in wrapper
    sys.stdout.write = output
AttributeError: 'file' object attribute 'write' is read-only
```
I'll revisit this later.

--- a/tests.py
+++ b/tests.py
@@ -25,7 +25,10 @@ def stdout_capture(func):
             self._value = u""
 
         def write(self, message):
-            self._value += message
+            try:
+                self._value += message.decode("utf-8")
+            except AttributeError:  # Already decoded
+                self._value += message
 
         @property
         def value(self):

--- a/tests.py
+++ b/tests.py
@@ -16,15 +16,15 @@ def stdout_capture(func):
     """Decorator to capture stdout during a test for testing the command line
     If you need to actually print to stdout during a test, use ._print
     """
-    sys_stdout_write = sys.stdout.write
+    sys_stdout = sys.stdout
 
     class MockWriter(object):
-        _print = sys_stdout_write
+        _print = sys_stdout.write
 
         def __init__(self):
             self._value = u""
 
-        def __call__(self, message):
+        def write(self, message):
             self._value += message
 
         @property
@@ -39,13 +39,13 @@ def stdout_capture(func):
     def wrapper(obj):
         output = MockWriter()
         # Override stdout.write with the mock
-        sys.stdout.write = output
+        sys.stdout = output
 
         # Pass the mock to the test
         result = func(obj, output)
 
         # Restore stdout
-        sys.stdout.write = sys_stdout_write
+        sys.stdout = sys_stdout
         return result
     return wrapper
 

